x-pack/plugins/security_solution/public/common/containers/dashboards/use_security_dashboards.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { MouseEventHandler } from 'react';
+import React, { useState, useEffect, useMemo, useCallback } from 'react';
+import type { EuiBasicTableColumn } from '@elastic/eui';
+import { EuiLink } from '@elastic/eui';
+import type { SavedObjectAttributes } from '@kbn/securitysolution-io-ts-alerting-types';
+import type { SavedObjectsClientContract, SavedObject } from '@kbn/core/public';
+import { useKibana } from '../../lib/kibana';
+import * as i18n from './translations';
+
+export interface DashboardTableItem extends SavedObject<SavedObjectAttributes> {
+  title?: string;
+  description?: string;
+}
+
+const SECURITY_TAG_NAME = 'security' as const;
+const EMPTY_DESCRIPTION = '-' as const;
+
+const getSecurityDashboardItems = async (
+  savedObjectsClient: SavedObjectsClientContract
+): Promise<DashboardTableItem[]> => {
+  if (savedObjectsClient) {
+    const tagResponse = await savedObjectsClient.find<SavedObjectAttributes>({
+      type: 'tag',
+      searchFields: ['name'],
+      search: SECURITY_TAG_NAME,
+    });
+
+    const tagId = tagResponse.savedObjects[0]?.id;
+
+    if (tagId) {
+      const dashboardsResponse = await savedObjectsClient.find<SavedObjectAttributes>({
+        type: 'dashboard',
+        hasReference: { id: tagId, type: 'tag' },
+      });
+
+      return dashboardsResponse.savedObjects.map((item) => ({
+        ...item,
+        title: item.attributes.title?.toString() ?? undefined,
+        description: item.attributes.description?.toString() ?? undefined,
+      }));
+    }
+  }
+  return [];
+};
+
+export const useSecurityDashboardsTableItems = () => {
+  const [dashboardItems, setDashboardItems] = useState<DashboardTableItem[]>([]);
+
+  const {
+    savedObjects: { client: savedObjectsClient },
+  } = useKibana().services;
+
+  useEffect(() => {
+    let ignore = false;
+    const fetchDashboards = async () => {
+      const items = await getSecurityDashboardItems(savedObjectsClient);
+      if (!ignore) {
+        setDashboardItems(items);
+      }
+    };
+
+    fetchDashboards();
+    return () => {
+      ignore = true;
+    };
+  }, [savedObjectsClient]);
+
+  return dashboardItems;
+};
+
+export const useDashboardsTableColumns = (): Array<EuiBasicTableColumn<DashboardTableItem>> => {
+  const {
+    application: { navigateToUrl },
```

Comment:
Yup, I used this one because we are navigating outside Security here, but you're right, using `useNavigateTo` and passing the url parameter will also work. I'll change it